test/extensions/filters/http/aws_lambda/aws_lambda_filter_test.cc

```diff
@@ -347,54 +359,74 @@ TEST_F(AWSLambdaFilterTest, SignsDataSetByPreviousFilters) {
   // we will manage credentials callbacks ourselves for this test
   setupRoute(false, false, false, false, true);
 
-  Http::TestRequestHeaderMapImpl headers{{":method", "GET"},
+  Http::TestRequestHeaderMapImpl headers_1{{":method", "GET"},
                                          {":authority", "www.solo.io"},
                                          {":path", "/getsomething"}};
-  Buffer::OwnedImpl data("data");
-  Http::TestRequestTrailerMapImpl trailers;
+  Buffer::OwnedImpl data_1("data");
+  Http::TestRequestTrailerMapImpl trailers_1;
 
 
   StsConnectionPool::Context::Callbacks *callbackReference;
   EXPECT_CALL(*filter_config_, getCreds).WillRepeatedly(
-     [&](StsConnectionPool::Context::Callbacks *callbacks){
+     [&](StsConnectionPool::Context::Callbacks *callbacks) ->
+                                         StsConnectionPool::Context*{
         callbackReference = callbacks;
+        // context is used for evaluating if we are in async
+        // only other use is cancel so just use stub.
+        return new StsContextStub();
```

Comment:
this will leak memory; while not really a problem as this is a test, it will get annoying when we do asan builds to debug future memory issues.
i suggest having it on the stack (near `callbackReference` and returning a pointer to it?)

Reply:
Sure thing. Asan complained about the unused mock but didnt claim that this leaked.